src/T2DMap.cpp

```diff
@@ -1025,13 +1025,13 @@ void T2DMap::paintEvent(QPaintEvent* e)
                     customLinePen.setCapStyle(Qt::RoundCap);
                     customLinePen.setJoinStyle(Qt::RoundJoin);
 
-                    if (_style == "solid line") {
+                    if (_style == tr("solid line")) {
                         customLinePen.setStyle(Qt::SolidLine);
-                    } else if (_style == "dot line") {
+                    } else if (_style == tr("dot line")) {
                         customLinePen.setStyle(Qt::DotLine);
-                    } else if (_style == "dash line") {
+                    } else if (_style == tr("dash line")) {
                         customLinePen.setStyle(Qt::DashLine);
-                    } else if (_style == "dash dot line") {
+                    } else if (_style == tr("dash dot line")) {
```

Comment:
Damn! This can't be translated as these strings are also used programatically internally to identify the state, right @SlySven ?! I would love to split the string displayed from internal state variables. In fact, the state should be sufficient as a mere number 1-4 which will be translated into strings for display, depending on the language chosen. Then again, my C++ is not yet versatile enough to implement said split. Please help!

For comparison see code around/before this area here: https://github.com/Mudlet/Mudlet/blob/466219c8fb75e16f629ef472bd957fc6b74462b6/src/TLuaInterpreter.cpp#L8159

I would really love to translate the strings displayed (compare screenshot below) but if no advice is given, I must again remove these for now, in order not to break any functionality.

Please in the future, never use strings for both display and internal state, ok? Thanks!

![image](https://user-images.githubusercontent.com/117238/47615479-980e3500-daaf-11e8-8c8d-6d192d3a6d5f.png)

Reply:
You will find instances of this sadly, dating back to before internationalisation was even thought about. Exit directions in some parts of the `TRoom` data elements are another which even now exercise coding muscles.

As for the custom line type it would be well worth changing the `(QMap<QString, QString>) TRoom::customLinesStyle` to be `QMap<QString, Qt::PenStyle> TRoom::customLinesStyle` and storing the corresponding `Qt::PenStyle` enum values into the data item for each entry in the `QComboBox`'s - then you can use the `QComboBox::currentData()` method - possibly with a `static_cast<Qt::PenStyle>(...)` or what have you to compare/set the Custom Line style directly - and it would probably simplify slightly the `T2DMap::paintEvent()` code a tiny bit...! :open_mouth:

Reply:
This seems out of my league for now. Need to postpone those translations and handle as a seperate issue.

Reply:
Yep agreed, just get the ones you're comfortable with.

Reply:
:stopwatch: I am rolling a fix for this *misfeature* and hope to have it ready for review in next 24 Hours...

Reply:
Now available as PR #2106 :innocent: